Makefile

```diff
@@ -1,15 +1,41 @@
+#!/usr/bin/env bash
+
 # Application
-APP_ROOT := $(CURDIR)
+APP_ROOT := $(abspath $(lastword $(MAKEFILE_LIST))/..)
 APP_NAME := twitcher
 
-# Anaconda
-CONDA := $(shell command -v conda 2> /dev/null)
-ANACONDA_HOME := $(shell conda info --base 2> /dev/null)
-CONDA_ENV := $(APP_NAME)
+# Conda
+CHECK      ?= shell command -v
+CONDA_HOME ?= $($(CHECK) conda info --base 2> /dev/null)
+CONDA_ENV  ?= $(APP_NAME)
+CONDA_BIN  := $(CONDA_HOME)/bin/conda
+CONDA_TARGET_PREFIX := readlink -e "$(CONDA_HOME)/envs/$(CONDA_ENV)" 2> /dev/null
+CONDA_ACTUAL_PREFIX := readlink -e "${CONDA_PREFIX}" 2> /dev/null
+# don't activate and update conda env if it already is activated
+# (avoids long conda 'solving environment' time)
+ifeq ($(shell $(CONDA_TARGET_PREFIX)),$(shell $(CONDA_ACTUAL_PREFIX)))
```

Comment:
This check is "true" when there is both no conda env and no `CONDA_PREFIX` ... not what we want.

Reply:
On my side, this works as intended. 
On the other hand, the previously used commands were failing on my side : 
```
CONDA := $(shell command -v conda 2> /dev/null)
ANACONDA_HOME := $(shell conda info --base 2> /dev/null)
```

What values are you getting (use `make debug`) ?
I would like to investigate to find a method that works for everyone.
I believe that `conda` variants can be very intricate depending on which one it is (anaconda, miniconda, etc.)

Reply:
On ubuntu docker image I get:
```
$ make debug
Following variables are used:
  SHELL:                         /bin/sh
  CHECK:                         shell command -v
  APP_ROOT:                      /opt/twitcher
  APP_NAME:                      twitcher
  BUMP_XARGS:                    --verbose --allow-dirty
  CONDA_HOME:                    
  CONDA_BIN:                     /bin/conda
  CONDA_ENV:                     twitcher
  CONDA_MSG:                     Conda environment already activated
  CONDA_CMD:                     echo Conda environment already activated;
  CONDA_TARGET_PREFIX [literal]: readlink -e /envs/twitcher 2> /dev/null
  CONDA_ACTUAL_PREFIX [called]:  
  CONDA_TARGET_PREFIX [literal]: readlink -e  2> /dev/null
  CONDA_ACTUAL_PREFIX [called]:  
  DOCKER_TAG:                    birdhouse/twitcher:0.4.0
```

Reply:
miniconda is installed:
```
which conda
/opt/anaconda/bin/conda
```

Reply:
If you run `make CHECK="command -v" debug`, does your `CONDA_HOME` display a valid path instead of blank?

I suspect that the `$(shell $(...))` syntax might be causing the issue. 
If that's the case, `ifeq (`$(CONDA_TARGET_PREFIX)`,`$(CONDA_ACTUAL_PREFIX)`)` instead of the highlighted code will probably do the trick.

I get the same output for both on my side.

Reply:
I run this on my ubuntu docker container ... but does not change it:
```
$ make CHECK="command -v" debug
Following variables are used:
  SHELL:                         /bin/sh
  CHECK:                         command -v
  APP_ROOT:                      /opt/twitcher
  APP_NAME:                      twitcher
  BUMP_XARGS:                    --verbose --allow-dirty
  CONDA_HOME:                    
  CONDA_BIN:                     /bin/conda
  CONDA_ENV:                     twitcher
  CONDA_MSG:                     Conda environment already activated
  CONDA_CMD:                     echo Conda environment already activated;
  CONDA_TARGET_PREFIX [literal]: readlink -e /envs/twitcher 2> /dev/null
  CONDA_ACTUAL_PREFIX [called]:  
  CONDA_TARGET_PREFIX [literal]: readlink -e  2> /dev/null
  CONDA_ACTUAL_PREFIX [called]:  
  DOCKER_TAG:                    birdhouse/twitcher:0.4.0
```

Reply:
Both on ubuntu and on macos I'm using bash. Do you have a different shell which would cause a difference?

Reply:
I think it is because of miniconda alternative configs that define `conda.csh` scripts to do some pre-processing steps before activating the environment. This causes the conda "bin" to actually point to another script which messes up commands like `which`.
If have found that `CONDA_EXE` points to the real conda bin path though. 
So if it does on your side too in both situations when env is (de)activated, I could adapt the script to use this variable.

Reply:
I have `CONDA_EXE` only when I run `conda activate`:
```
$ conda activate
$ env | grep -i conda
$ CONDA_EXE=/usr/local/anaconda3/bin/conda
```

But the env is not defined when I just put the conda-bin in `PATH`.

Does `which` help?
https://github.com/bird-house/pyramid-phoenix/blob/5c5a4f73778c31abdee13ce004dd651b43f24247/docs/Makefile#L11